transcom/mymove
# [MB-16183] Add service member authorization check for ListMTOShipments

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16183)

## Summary

This pull request adds an authorization check for customer users, when hitting the `/moves/:moveId/mto_shipments` endpoint with a GET request, to verify that they are the customer who is attached to the move and thereby have authorization to view the shipments.

Since the service object that fetches the list of shipments for a move is used by both the internal and GHC API endpoints, the authorization check is conditional on this having come from the customer application.

Service object, handler, and integration tests are added or updated as necessary.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Verify that all of the server tests pass.
2. Log in to the customer-facing application as a customer user, and create a move with at least one shipment. Record the move ID (which is a UUID that will display as a fragment of the in the URL bar in the browser)
3. Using postman, make a request to the `/moves/:moveID/mto_shipments` endpoint while authenticated as a different customer user, and verify that a 404 is returned.

### Backend
